lib/resty/mlcache/ipc.lua

```diff
@@ -113,11 +105,21 @@ function _M:broadcast(channel, data)
         return nil, "failed to increment index: " .. err
     end
 
-    local ok, err = self.dict:set(idx, marshalled_event)
+    local ok, err, forcible = self.dict:set(idx, marshalled_event)
     if not ok then
         return nil, "failed to insert event in shm: " .. err
     end
 
+    if forcible then
+        -- take note that eviction has started
+        -- we repeat this flagging to avoid this key from ever being
+        -- evicted itself
```

Comment:
A better approach here would be if `shm:incr()` accepted `flags`, or maybe even an `shm:set_flags()` API. In the absence of a solution to rely on flags, we rely on an extraneous key, and on it never reaching the LRU tail.